src/beamlime/kafka/message_adapter.py

```diff
@@ -0,0 +1,131 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from dataclasses import replace
+from typing import Any, Generic, Protocol, TypeVar
+
+import scipp as sc
+import streaming_data_types
+import streaming_data_types.exceptions
+from streaming_data_types import dataarray_da00, eventdata_ev44
+
+from ..core.message import Message, MessageKey, MessageSource
+from ..handlers.monitor_data_handler import MonitorEvents
+from .scipp_da00_compat import da00_to_scipp
+
+T = TypeVar('T')
+U = TypeVar('U')
+V = TypeVar('V')
+
+
+class KafkaMessage(Protocol):
+    """Simplified Kafka message interface for testing purposes."""
+
+    def error(self) -> Any | None:
+        pass
+
+    def value(self) -> bytes:
+        pass
+
+    def topic(self) -> str:
+        pass
+
+
+def message_schema(msg: KafkaMessage) -> str | None:
+    """
+    Extracts the schema from a Kafka message by the streaming_data_types library.
+    """
+    if msg.error() is not None or len(msg.value()) < 8:
+        return None
+    return msg.value()[4:8].decode()
+
+
+class FakeKafkaMessage(KafkaMessage):
+    def __init__(self, value: bytes, topic: str):
+        self._value = value
+        self._topic = topic
+
+    def error(self) -> Any | None:
+        return None
+
+    def value(self) -> bytes:
+        return self._value
+
+    def topic(self) -> str:
+        return self._topic
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, FakeKafkaMessage):
+            return False
+        return self._value == other._value and self._topic == other._topic
+
+
+class MessageAdapter(Protocol, Generic[T, U]):
+    def adapt(self, message: T) -> U:
+        pass
```

Comment:
I find it a bit cumbersome to follow this `adapt` chain.
Can we just use sciline pipeline here later...? So that we can also visualize the chain.

Reply:
I don't think Sciline is useful here. There are just two adapters in a chain.

Reply:
Then why do we need a chain of adapters at all...? We can just handle it at once.